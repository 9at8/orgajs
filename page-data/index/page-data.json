{"componentChunkName":"component---src-pages-index-js","path":"/","webpackCompilationHash":"2ffcc636619e4e20966b","result":{"pageContext":{"isCreatedByStatefulCreatePages":true}}}