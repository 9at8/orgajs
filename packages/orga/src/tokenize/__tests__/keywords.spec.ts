import { tokenize } from '../index'

describe('tokenize keywords', () => {

  it('knows keywords', () => {
    expect(tokenize('#+KEY: Value').all()).toMatchSnapshot()
    expect(tokenize('#+KEY: Another Value').all()).toMatchSnapshot()
    expect(tokenize('#+KEY: value : Value').all()).toMatchSnapshot()
  })

  it('knows these are not keywords', () => {
    expect(tokenize('#+KEY : Value').all()).toMatchSnapshot()
    expect(tokenize('#+KE Y: Value').all()).toMatchSnapshot()
  })

})
