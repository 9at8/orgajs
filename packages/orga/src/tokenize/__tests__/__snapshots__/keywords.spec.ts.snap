// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize keywords knows keywords 1`] = `
Array [
  Object {
    "data": Object {
      "key": "KEY",
      "value": "Value",
    },
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "keyword",
  },
]
`;

exports[`tokenize keywords knows keywords 2`] = `
Array [
  Object {
    "data": Object {
      "key": "KEY",
      "value": "Another Value",
    },
    "position": Object {
      "end": Object {
        "column": 20,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "keyword",
  },
]
`;

exports[`tokenize keywords knows keywords 3`] = `
Array [
  Object {
    "data": Object {
      "key": "KEY",
      "value": "value : Value",
    },
    "position": Object {
      "end": Object {
        "column": 20,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "keyword",
  },
]
`;

exports[`tokenize keywords knows these are not keywords 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize keywords knows these are not keywords 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;
